internal/archive/tar/tar.go

```diff
@@ -0,0 +1,70 @@
+package tar
+
+import (
+	"archive/tar"
+	"bytes"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/saucelabs/saucectl/internal/sauceignore"
+)
+
+// Resource archives the resource and exclude files and folders based on sauceignore logic.
+func Resource(src string, matcher sauceignore.Matcher) (io.Reader, error) {
+	bb := new(bytes.Buffer)
+	w := tar.NewWriter(bb)
+	defer w.Close()
+
+	walker := func(file string, fileInfo os.FileInfo, err error) error {
+		// Only will be applied if we have .sauceignore file and have patterns to exclude files and folders
+		if matcher.Match(strings.Split(file, string(os.PathSeparator)), fileInfo.IsDir()) {
+			return nil
+		}
+
+		header, err := tar.FileInfoHeader(fileInfo, file)
+		if err != nil {
+			return err
+		}
+
+		relFilePath := file
+		if filepath.IsAbs(src) {
+			// copy temp files
```

Comment:
Why is this necessary? 🤔 This block here doesn't really explain what it's trying to do 🤔